importing raise_error and molutil

loriab opened the pull request:
- [x] I'm not wholly convinced, but add `raise_error` for default to `which` and `which_import` after discussion in https://github.com/MolSSI/QCEngine/pull/65
- [x] Replace deprecated `find_loader` with `find_spec`.
- [x] Q: `ModuleNotFoundError` is _not_ right for `which` but what is?
- [x] Move numpy alignment to molutil

dgasmith commented on qcelemental/tests/test_importing.py:
```diff
@@ -23,6 +25,13 @@ def test_which_import_f_bool():
+    with pytest.raises(ModuleNotFoundError) as e:
+        qcel.util.which_import('evilpint', raise_error=True)
+
+    assert "'evilpint' not found" in str(e)
```
Hmm... no such thing.

dgasmith commented on qcelemental/util/importing.py:
```diff
@@ -40,12 +48,20 @@ def which(command, *, return_bool=False):
     ans = shutil.which(command, mode=os.F_OK | os.X_OK, path=lenv['PATH'])
 
+    if raise_error and ans is None:
+        raise ModuleNotFoundError(f"Command '{command}' not found in envvar PATH.")
```
I think you convinced me of this one where we should always add custom messages. What if we added:

`"Command '{command}' not found in envvar PATH. {if install else ''}"`

or similar here? Would hide a bit more logic.

loriab replied:
Where `install` is a message passed into the fn? I decided against that once, but not strenuously. Fine by me.

loriab commented:
this is RTG

dgasmith commented:
Squish LGTM commits?

loriab commented:
squish the whole thing, imo

loriab commented:
Fixed some `os.pathsep` and added an `env` to `which` to help code with custom envvar. Not sure I did it the best way. Here's a context manager: https://github.com/laurent-laporte-pro/stackoverflow-q2059482/blob/master/demo/environ_ctx.py

dgasmith commented:
LGTM
